tests/mnle_test.py

```diff
@@ -0,0 +1,199 @@
+
+    prior = MultipleIndependent(
+        [
+            Gamma(torch.tensor([1.0]), torch.tensor([0.5])),
+            Beta(torch.tensor([2.0]), torch.tensor([2.0])),
+        ],
+        validate_args=False,
+    )
+
+    num_simulations = 2000
+    num_samples = 1000
+    theta = prior.sample((num_simulations,))
+    x = mixed_simulator(theta)
+
+    # MNLE
+    trainer = MNLE(prior)
+    trainer.append_simulations(theta, x).train()
+    posterior = trainer.build_posterior()
+
+    mcmc_kwargs = dict(
+        num_chains=10,
+        warmup_steps=100,
+        method="slice_np_vectorized",
+        init_strategy="proposal",
+    )
+
+    for num_trials in [10]:
+        theta_o = prior.sample((1,))
+        x_o = mixed_simulator(theta_o.repeat(num_trials, 1))
+
+        # True posterior samples
+        transform = mcmc_transform(prior)
+        true_posterior_samples = MCMCPosterior(
+            PotenfialFunctionProvider(prior, atleast_2d(x_o)),
+            theta_transform=transform,
+            proposal=prior,
+            **mcmc_kwargs,
+        ).sample((num_samples,), show_progress_bars=False)
+
+        posterior = trainer.build_posterior(prior=prior, sample_with=sampler)
+        posterior.set_default_x(x_o)
+        if sampler == "vi":
+            posterior.train()
+
+        mnle_posterior_samples = posterior.sample(
+            sample_shape=(num_samples,),
+            show_progress_bars=False,
+            **mcmc_kwargs if sampler == "mcmc" else {},
+        )
+
+        check_c2st(
+            mnle_posterior_samples,
+            true_posterior_samples,
+            alg=f"MNLE with {sampler}",
+        )
+
+
+class PotenfialFunctionProvider(BasePotential):
+    """Returns potential function for reference posterior of a mixed likelihood."""
+
+    def __init__(self, prior, x_o, device="cpu"):
+        super().__init__(prior, x_o, device)
+
+    def __call__(self, theta, track_gradients: bool = True):
+
+        theta = atleast_2d(theta)
+
+        with torch.set_grad_enabled(track_gradients):
+            iid_ll = self.iid_likelihood(theta)
+
+        return iid_ll + self.prior.log_prob(theta)
+
+    def iid_likelihood(self, theta: torch.Tensor) -> torch.Tensor:
+        """Returns the likelihood summed over a batch of i.i.d. data."""
+
+        lp_choices = torch.stack(
+            [
+                Binomial(probs=th.reshape(1, -1)).log_prob(self.x_o[:, 1:])
+                for th in theta[:, 1:]
+            ],
+            dim=1,
+        )
+
+        lp_rts = torch.stack(
+            [
+                InverseGamma(
+                    concentration=2 * torch.ones_like(beta_i), rate=beta_i
+                ).log_prob(self.x_o[:, :1])
+                for beta_i in theta[:, :1]
+            ],
+            dim=1,
+        )
+
+        joint_likelihood = (lp_choices + lp_rts).reshape(
+            self.x_o.shape[0], theta.shape[0]
+        )
+
+        return joint_likelihood.sum(0)
+
+    def allow_iid_x(self):
```

Comment:
For consistency, could you use [this format](https://github.com/mackelab/sbi/blob/78cc11e8d194c4d7af83eb9bab4b4036536416d4/sbi/inference/potentials/likelihood_based_potential.py#L47-L48) for `allow_iid_x`?